vm/compiler/src/ledger/map/mod.rs

```diff
@@ -119,3 +110,34 @@ pub trait MapRead<
     ///
     fn values(&'a self) -> Self::Values;
 }
+
+/// This macro executes the given block of operations as a new atomic write batch IFF there is no
+/// atomic write batch in progress yet. This ensures that complex atomic operations consisting of
+/// multiple lower-level operations - which might also need to be atomic if executed individually -
+/// are executed as a single large atomic operation regardless.
+#[macro_export]
+macro_rules! atomic_write_batch {
+    ($self:expr, $ops:block) => {
+        // Check if an atomic batch write is already in progress.
+        let is_part_of_atomic_batch = $self.is_atomic_in_progress();
+
+        // Start an atomic batch write operation IFF it's not already part of one.
+        if !is_part_of_atomic_batch {
+            $self.start_atomic();
+        }
+
+        // Wrap the operations in a closure to know if any of them has failed.
```

Comment:
```suggestion
        // Wrap the operations in a closure to detect if any of them have failed.
```

Reply:
technically only one can fail at a time, but I'm ok with the wording adjustment :ok: